### PR TITLE
Use `haskell-flake`

### PR DESCRIPTION
**Pull request by HariAmoor-professional**
Uses https://github.com/srid/haskell-flake/pull/34

- [x] Fix infinite recursion bug

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -3,40 +3,24 @@
...
-              p.bar
-            ];
-            buildInputs = [
-              haskellPackages'.ghcid
-            ];
+      imports = [ haskell-flake.flakeModule ];
+      perSystem = { self', inputs', pkgs, system, ... }:
+        {
+          haskellProjects.default = {
+            haskellPackages = pkgs.haskell.packages.ghc924;
```
Comment (srid) [inline flake.nix]: You want to remove this, and let haskell-flake use the default package set from nixpkgs (for caching).

File: flake.nix [hunk 2]
```diff
@@ -3,40 +3,24 @@
...
+            root = ./.;
```
Comment (srid) [inline flake.nix]: Remove `root` (unused)

File: flake.nix [hunk 3]
```diff
@@ -3,40 +3,24 @@
...
+            overrides = _: _: { };
+            buildTools = { };
```
Comment (srid) [inline flake.nix]: This needs `ghcid`

File: flake.nix (hunk 1 above)
Comment (HariAmoor-professional) [reply flake.nix]: Done

File: flake.nix (hunk 2 above)
Comment (HariAmoor-professional) [reply flake.nix]: Done

File: flake.nix (hunk 3 above)
Comment (HariAmoor-professional) [reply flake.nix]: As discussed, it is not needed

**Review by srid — APPROVED** [review]

**Pull request merged**